src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -219,9 +221,6 @@ CHIP_ERROR UDPEndPointImplLwIP::SendMsgImpl(const IPPacketInfo * pktInfo, System
 
     ip_addr_copy(mUDP->local_ip, boundAddr);
 
-    // Unlock LwIP stack
-    UNLOCK_TCPIP_CORE();
```

Comment:
This is changing when the lock is released, no?  It would be better to have a scope around the actual code we want the lock held in.

Reply:
Added the scope around code which require locking.